gemini/evaluation/evaluating_prompts_at_scale_with_gemini_batch_prediction_api.ipynb

```diff
@@ -0,0 +1,1011 @@
+    "outputId": "45411834-7bee-4d00-8e60-6541df113b2d"
+   },
+   "outputs": [],
+   "source": [
+    "def download_blob_into_memory(bucket_name, blob_name):\n",
+    "    \"\"\"Downloads a blob from GCS into memory.\"\"\"\n",
+    "    bucket = storage_client.bucket(bucket_name)\n",
+    "\n",
+    "    blob = bucket.blob(blob_name)\n",
+    "    contents = blob.download_as_bytes()\n",
+    "    return contents\n",
+    "\n",
+    "\n",
+    "def classify_image(model_id, prompt, bucket_name, blob_name):\n",
+    "    \"\"\"Classifies an image.\"\"\"\n",
+    "    model = GenerativeModel(model_id)\n",
+    "    contents = download_blob_into_memory(bucket_name, blob_name)\n",
+    "    encoded_image = base64.b64encode(contents).decode(\"utf-8\")\n",
+    "    image_content = Part.from_data(\n",
+    "        data=base64.b64decode(encoded_image), mime_type=\"image/jpeg\"\n",
+    "    )\n",
+    "    contents = [prompt, image_content]\n",
+    "    response = model.generate_content(contents)\n",
+    "    return response\n",
+    "\n",
+    "\n",
+    "blob_name = get_filepaths(files_table=FILES_TABLE)[1]\n",
+    "response = classify_image(\n",
+    "    model_id=MODEL_ID, prompt=prompt, bucket_name=BUCKET_NAME, blob_name=blob_name\n",
+    ")\n",
+    "print(f\"blob_name: {blob_name}\")\n",
+    "print(f\"response: {response.text}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "Q79hx4JINXZh"
+   },
+   "source": [
+    "### Create a New Line JSON file applying the prompt to each of the images\n",
+    "In this section, also an `evaluation_id` variable is created to identify the execution run."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 85,
+   "metadata": {
+    "id": "_Dn4SStFOBak"
+   },
+   "outputs": [],
+   "source": [
+    "def create_newline_json_file(\n",
+    "    prompt, evaluation_ts, evaluation_id, file_name, bucket_name, images_uri\n",
+    "):\n",
+    "    \"\"\"Creates a newline delimited JSON file.\"\"\"\n",
+    "    with open(file_name, \"w\") as outfile:\n",
+    "        for image_uri in images_uri:\n",
+    "            contents = download_blob_into_memory(bucket_name, image_uri)\n",
+    "            encoded_image = base64.b64encode(contents).decode(\"utf-8\")\n",
+    "            request = {\n",
+    "                \"contents\": [\n",
+    "                    {\n",
+    "                        \"role\": \"user\",\n",
+    "                        \"parts\": [\n",
+    "                            {\"text\": prompt},\n",
+    "                            {\n",
+    "                                \"inlineData\": {\n",
+    "                                    \"mimeType\": \"image/jpeg\",\n",
+    "                                    \"data\": encoded_image,\n",
+    "                                }\n",
+    "                            },\n",
+    "                        ],\n",
+    "                    }\n",
+    "                ]\n",
+    "            }\n",
```

Comment:
Same as before, this should be able to load from GCS URIs (unless there's a limitation with the batch api?)

Reply:
Batch prediction doesn't support the fileData field for Gemini. See https://cloud.google.com/vertex-ai/generative-ai/docs/multimodal/batch-prediction-gemini#prepare_your_inputs